src/api/spec/models/project/update_from_xml_command_spec.rb

```diff
@@ -200,42 +199,68 @@
+              </project>
+            EOF
+          )
+          Project::UpdateFromXmlCommand.new(project).send(:update_repositories, xml_hash, false)
+        end
 
-      it "updates download repositories of a repository" do
-        expect(repository_1.download_repositories).to be_empty
+        it "updates download repositories of a repository" do
+          expect(repository_1.download_repositories).to be_empty
 
-        dod_repo = project.repositories.find_by(name: "dod_repo")
-        expect(dod_repo).not_to be_nil
-        expect(dod_repo.download_repositories.count).to eq 1
-      end
+          dod_repo = project.repositories.find_by(name: "dod_repo")
+          expect(dod_repo).not_to be_nil
+          expect(dod_repo.download_repositories.count).to eq 1
+        end
 
-      it "updates download_repository attributes" do
-        download_repository = project.repositories.find_by(name: "dod_repo").download_repositories.first
-        expect(download_repository.arch).to eq "i586"
-        expect(download_repository.repotype).to eq "rpmmd"
-        expect(download_repository.url).to eq "http://opensuse.org"
-        expect(download_repository.archfilter).to eq "i586, noarch"
-        expect(download_repository.masterurl).to eq "http://master.opensuse.org"
-        expect(download_repository.mastersslfingerprint).to eq "my_fingerprint"
-        expect(download_repository.pubkey).to eq "my_pubkey"
+        it "updates download_repository attributes" do
+          download_repository = project.repositories.find_by(name: "dod_repo").download_repositories.first
+          expect(download_repository.arch).to eq "i586"
+          expect(download_repository.repotype).to eq "rpmmd"
+          expect(download_repository.url).to eq "http://opensuse.org"
+          expect(download_repository.archfilter).to eq "i586, noarch"
+          expect(download_repository.masterurl).to eq "http://master.opensuse.org"
+          expect(download_repository.mastersslfingerprint).to eq "my_fingerprint"
+          expect(download_repository.pubkey).to eq "my_pubkey"
+        end
+      end
+      context "invalid usecase" do
+        before do
+          @xml_hash = Xmlhash.parse(
+            <<-EOF
+              <project name="#{project.name}">
+                <repository name="repo_1" />
+                <repository name="dod_repo">
+                  <download arch='i586' url='http://opensuse.org' repotype='rpm-md'>
+                    <archfilter>i586, noarch</archfilter>
+                    <master url='http://master.opensuse.org' sslfingerprint='my_fingerprint'/>
+                    <pubkey>my_pubkey</pubkey>
+                  </download>
+                  <arch>i586</arch>
+                </repository>
+              </project>
+            EOF
+          )
+        end
+        it "barfs" do
+          expect { Project::UpdateFromXmlCommand.new(project).send(:update_repositories, @xml_hash, false) }.to raise_error(
```

Comment:
All problems in computer science can be solved by another level of indirection... ;-)